docs/okeps/okep-5259-no-overlay.md

```diff
@@ -0,0 +1,598 @@
+    kind: FRRConfiguration
+    metadata:
+      name: route-generated-blue
+      namespace: frr-k8s-system
+    spec:
+      bgp:
+        routers:
+        - asn: 64512
+          neighbors:
+          - address: 10.89.0.37
+            asn: 64512
+            disableMP: true
+            toAdvertise:
+              allowed:
+                prefixes:
+                # the node subnet of this node
+                - 10.244.1.0/24
+          prefixes:
+          - 10.244.1.0/24
+    ```
+
+#### Create A ClusterUserDefinedNetwork in No-overlay Mode
+
+For user defined networks, to ensure the traffic isolation, no-overlay mode can
+only be enabled when the route advertisement is configured in a VRF-Lite mode.
+The cross node east-west traffic in this user defined network will be isolated
+by the VRFs in each node and the VLANs of the provider network. VRF-Lite is only
+available when using Local Gateway Mode, thus no-overlay mode user defined
+networks is also only available when using Local Gateway Mode.
+
+Here's a configuration example:
+
+1. A cluster admin wants to enable no-overlay mode for the blue network. The
+   cluster admin has to do all host modifications necessary via NMState to
+   enslave a VLAN interface to the VRF `blue` on each node before creating the
+   following ClusterUserDefinedNetwork CR.
+
+    ```yaml
+    apiVersion: k8s.ovn.org/v1
+    kind: ClusterUserDefinedNetwork
+    metadata:
+      name: blue
+    spec:
+        namespaceSelector:
+          matchExpressions:
+          - key: kubernetes.io/metadata.name
+            operator: In
+            values: ["ns1", "ns2"]
+        network:
+          topology: Layer3
+          layer3:
+            role: Primary
+            # The UDN MTU shall not be larger than the provider network's MTU.
+            mtu: 1500
+            subnets:
+            - cidr: 10.10.0.0/16
+              hostSubnet: 24
+            encapsulation: None
+    ```
+
+1. The cluster admin has created an FRRConfiguration CR to peer with a external
+   BGP router `182.18.0.5` on the blue VLANs.
+
+    ```yaml
+    apiVersion: frrk8s.metallb.io/v1beta1
+    kind: FRRConfiguration
+    metadata:
+      name: vpn-blue
+      namespace: frr-k8s-system
+      labels:
+        network: vpn-blue
+    spec:
+      bgp:
+        routers:
+        - asn: 64512
+          vrf: blue
+          neighbors:
+          - address: 182.18.0.5
+            asn: 64512
+            disableMP: true
+            holdTime: 1m30s
+            keepaliveTime: 30s
+            passwordSecret: {}
+            port: 179
+            toAdvertise:
+              allowed:
+                mode: filtered
+            toReceive:
+              allowed:
+                mode: filtered
+    ```
+
+1. The cluster admin has to advertise this pod network of the CUDN by BGP. The
+   targetVRF is set to auto, meaning routes to the podNetworks will only be
+   advertised within the corresponding VRF.
```

Comment:
So if I am understanding this right, OVNK will create a RouteAdvertisements for the default network, but not for CUDNs?

Reply:
I've updated the steps. Users shall create the RA.